### ISSUE TITLE
[Suggestion] Global economics

### ISSUE THREAD

**Issue by OverSavior** [tsunamayo/Skywanderers#1600]
For the last 5 or so weeks I was playing a game called Prosperous Universe. It's a space-themed browser economic simulator. I never thought I would enjoy this kind of gameplay it offers, but it appeared to be really fun.

I think that Skywanderers could benefit a lot from similar mechanics, too. While I know that this game is supposed to be mainly about exploration, I still think it would dramatically increase the player base if something like that was possible in game. Let me tell you a bit more about PrUn first. In PrUn, you own a space company. It can do anything from extracting gasses, ores or liquids from planets to manufacturing advanced technical pieces or even building entire ships. There's also fuel production btw. 

The thing that I found to be the most interesting is that in that game there's no bots. Absolutely everything on the market is produced by players. Say, some player extracts ore, smelts it into metal and sells as is. Then some other player buys that metal and use it to construct prefabs. Then another player uses these prefabs to build another base so he can build more ore extractors. All of them benefit from this, as well as the person who produces supplies such as food and drinking water, which everyone need for their production to operate.

The most interesting part here is trading. How players have to balance their prices for others to want to buy from them. How lack of something creates demand and how this demand affects prices of everything else. This is very complicated and I could go deeper into details if you'd like to, but for now let's stop here.

How I imagine this could be applied to Skw world is by creating a global world with similar rules, where wast majority of resources would need to be supplied by players. For the market at least. Like, all blocks and systems could be either produced or purchased in market. Purchasing is a lot easier if you have money, but producing something could benefit one a lot.

If market and the game itself worked this way, this would create countless possibilities for people that would never end. The game wouldn't end on building a large ship or something. Everyone would have their role. Let's imagine that there's some system where you can buy fuel for a very low price. A bunch of players decided that being a pirate is a good idea. Because of their raids on transport vessels the fuel price suddenly increases, and increases a lot, because those who suffered and couldn't delived as much fuel to the market need to somehow cover their damage.

In such situation there would be at least 2 different ways to fix the problem. One would be to expand production or move it to other location (refinery can be either ship based or as a whole large orbital station). Another is to hire someone who could deal with pirates and either protect transport vessels or elliminate pirate threat entirely by destroying their base or supply ship.

This sure would be very hard to implement not only because of complicated mechanics, but also because it would need to be properly designed and balanced, but I think this could be worth a shot. Maybe it could be simplified, but I feel like even basic economics would bring a lot more possibilities to the game. Afterall, not everyone enjoys combat or space exploration. Some people would find their place in this economy niche. What are your thoughts on this?

**Comment by ZachZent** [comment]
Now I'm not too sure, but it sounds like a game like these requires a single shared server. Unless something changed I believe the plan is to have it be Minecraft like where you can start your own server. Plus I'm not sure how well ti will work when NPCs will be a large focus for trading, crewing, etc. Best keep it a static price, or better yet a star system based economy where a lack of one type of resource (say there is no ice planet) means that resource can be sold for more (ice costs more.) This allows for the creation of trade routes from one system with a surplus of X to one with a lack of X.

**Comment by ultrak2k** [comment]
> Now I'm not too sure, but it sounds like a game like these requires a single shared server. Unless something changed I believe the plan is to have it be Minecraft like where you can start your own server. Plus I'm not sure how well ti will work when NPCs will be a large focus for trading, crewing, etc. Best keep it a static price, or better yet a star system based economy where a lack of one type of resource (say there is no ice planet) means that resource can be sold for more (ice costs more.) This allows for the creation of trade routes from one system with a surplus of X to one with a lack of X.

I disagree - its very possible to have a not boring as shit dynamic galactic or semi galactic economy without mass multiplayer, even singleplayer, starsector for example does it STUNNINGLY. I think its difficult but not impossible and really important for the future of the game, as if we want a trade and factory gamestyle to be fun, we need a dyn-eco

Please tsuna, consider slowly implementing a dynamic economy, even if he starting one is basic, it can ramp up

**Comment by tsunamayo** [comment]
Sorry I havent time to read it yet...

**Comment by ultrak2k** [comment]
Oh, I also think we don't really need an mmo for a realistic dynamic economy , I forgot to mention this but even having NPC traders and automated/Automated with NPC factories can have the same effect. better then having everything manually done, fuck that. Factory gameplay and trade convoys the player can set up I assume would be a thing, since that's simply fleet mechanics + i think confirmed factory stuff, but ill put this out for tsuna to read

**Comment by Aboleth** [comment]
I think players may be able to do this by themselves via logic ingame, all that's needed is a way to get data from in-game logic outside of the game to be collated. Some people I know are doing this in spengies and it's working well.
